Sources/Confidence/ConfidenceValue.swift

```diff
@@ -0,0 +1,231 @@
+import Foundation
+
+public typealias ConfidenceStruct = [String: ConfidenceValue]
+
+public class ConfidenceValue: Equatable, Encodable {
+    private let value: ConfidenceValueInternal
+
+    public init(boolean: Bool) {
+        self.value = .boolean(boolean)
+    }
+
+    public init(string: String) {
+        self.value = .string(string)
+    }
+
+    public init(integer: Int64) {
+        self.value = .integer(integer)
+    }
+
+    public init(double: Double) {
+        self.value = .double(double)
+    }
+
+    public init(date: DateComponents) {
+        self.value = .date(date)
+    }
+
+    public init(timestamp: Date) {
+        self.value = .timestamp(timestamp)
+    }
```

Comment:
Is it better to use arg labels `date` and `timestamp` (i.e. the Confidence-types), like shown here, or rather use respectively `dateComponentList` and `dateList` (i.e. the accepted Swift types)?

The problem with the latter approach would be that only the documentation can explain to the user that a Confidence Timestamp is actually achieved by using the Date constructor

Reply:
I also like it more as it is. The one at fault here really is the Swift language that calls a seconds-precision timestamp `Date` instead of `Instant` :)